packages/web-components/src/components/masthead/top-nav.ts

```diff
@@ -21,18 +24,40 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  * @element dds-top-nav
  */
 @customElement(`${ddsPrefix}-top-nav`)
-class DDSTopNav extends StableSelectorMixin(BXHeaderNav) {
+class DDSTopNav extends StableSelectorMixin(HostListenerMixin(BXHeaderNav)) {
   /**
    * `true` to hide the divider.
    */
   @property({ type: Boolean, reflect: true, attribute: 'hide-divider' })
   hideDivider = false;
 
+  /**
+   * Hide top nav flag.
+   */
+  @internalProperty()
+  private _hideNav = false;
+
   static get stableSelector() {
     return `${ddsPrefix}--masthead__l0-nav`;
   }
 
   static styles = styles; // `styles` here is a `CSSResult` generated by custom WebPack loader
+
+  render() {
+    return this._hideNav ? html`` : super.render();
+  }
+
+  /**
+   * Handles toggle event from the search component.
+   *
+   * @param event The event.
+   */
+  // private _handleSearchToggle: (event: Event) => void;
+  @HostListener(`parentRoot:${DDSMastheadSearch.eventToggle}`)
```

Comment:
To support alternate (typically) component composed in masthead, this codebase makes event name configurable in receiving component side ([an example](https://github.com/carbon-design-system/carbon-web-components/blob/v1.5.0/src/components/ui-shell/side-nav.ts#L198-L200)). So please define `eventToggle` static property to each classes receiving the event:

```suggestion
  @HostListener(`parentRoot:eventToggle`)
```